pytorch_lightning/core/lightning.py

```diff
@@ -1146,8 +1174,77 @@ def backward(self, loss, optimizer, optimizer_idx):
                 loss.backward()
 
         """
-        if self.trainer.train_loop.automatic_optimization or self._running_manual_backward:
-            loss.backward(*args, **kwargs)
+        loss.backward(*args, **kwargs)
+
+    def _backward_wrapper(self, loss: Tensor, optimizer: Optimizer, optimizer_idx: int, *args, **kwargs) -> None:
+        automatic_optimization = self.trainer.train_loop.automatic_optimization
+
+        if automatic_optimization or self._running_manual_backward:
+
+            is_ddp = isinstance(self.trainer.model, LightningDistributedDataParallel)
+            never_skip = self.invalid_loss_strategy == InvalidLossStrategy.NEVER_SKIP
+
+            if is_ddp and never_skip and automatic_optimization:
+                self._backward_with_possible_nan_loss(loss, optimizer, optimizer_idx, *args, **kwargs)
+            else:
+                self.backward(loss, optimizer, optimizer_idx, *args, **kwargs)
+
+    def _backward_with_possible_nan_loss(self, loss: Tensor, optimizer: Optimizer, optimizer_idx: int, *args, **kwargs) -> None:
+        """
+        This function is used to handle the case when a loss in a ``DistributedDataParallel``
+        """
+        self.__check_invalid_loss(loss)
+
+        # prevent ddp grad synchronization
+        self.trainer.model.require_backward_grad_sync = False
+
+        if not torch.isnan(loss):
+            self.backward(loss, optimizer, optimizer_idx, *args, **kwargs)
+        else:
+            self.__create_zeros_gradients()
+
+        if not self.trainer.train_loop.should_accumulate():
+            self.__synchronize_gradients(optimizer)
+
+    def __check_invalid_loss(self, loss):
+        is_invalid = torch.isnan(loss).int()
+        self.trainer.accelerator_backend.sync_tensor(is_invalid)
+        self._invalid_loss_counts.append(is_invalid)
+        return False
+
+    def __create_zeros_gradients(self):
+        """
+        This function will generate zeros gradients when None for the first reduction
+        """
+        for p in self.parameters():
+            if p.requires_grad and p.grad is None:
+                p.grad = torch.zeros_like(p, device=self.device, dtype=torch.float)
```

Comment:
any reason why you are manually setting these?

```suggestion
                p.grad = torch.zeros_like(p)
```